src/sagas/PostWork.saga.test.ts

```diff
@@ -0,0 +1,177 @@
+import { Frost } from '@po.et/frost-client'
+import { delay } from 'redux-saga'
+import { takeLatest, call, put } from 'redux-saga/effects'
+import { describe } from 'riteway'
+
+import { Actions } from '../actions/index'
+import { PostWorkSaga, PostWork, redirectToWork } from './PostWork.saga'
+import { WorkAttributes } from '../interfaces/Props'
+import { Configuration } from '../configuration'
+
+const createWork = ({
+  name = 'name',
+  dateCreated = new Date().toISOString(),
+  datePublished = new Date().toISOString(),
+  text = 'test',
+  tags = 'test',
+  author = 'author'
+}: WorkAttributes = {
+  name: 'name',
+  dateCreated: new Date().toISOString(),
+  datePublished: new Date().toISOString(),
+  text: 'test',
+  tags: 'test',
+  author: 'author'
+}) => ({
+  name,
+  dateCreated,
+  datePublished,
+  text,
+  tags,
+  author
+})
+
+describe('PostWork Saga', async assert => {
+  const iterator = PostWorkSaga()()
+  
+  assert({
+    given: 'post work action',
+    should: 'handle post work request',
+    actual: iterator.next().value,
+    expected: takeLatest(Actions.PostWork.POST_WORK, PostWork),
+  })
+})
+
+describe('PostWork() Success', async assert => {
+  const token = 'test-token'
+  const work = createWork()
+  const workId = 'testWorkId'
+
+  const iterator = PostWork(Actions.PostWork.onPostWork({ work, token }))
+
+  assert({
+    given: 'post work action',
+    should: 'set loading status',
+    actual: iterator.next().value,
+    expected: put(Actions.LoadingPage.onLoadingOn()),
+  })
+
+  {
+    const frost = new Frost({ host: Configuration.frostApiUrl })
+    assert({
+      given: 'next step',
+      should: 'post work with frost',
+      actual: iterator.next().value,
+      expected: call([frost, frost.createWork], token, work),
+    })
+  }
+
+  assert({
+    given: 'next step',
+    should: 'clear loading',
+    actual: iterator.next({ workId }).value,
+    expected: put(Actions.LoadingPage.onLoadingFull())
+  })
+
+  assert({
+    given: 'next step',
+    should: 'set workId in notification bar',
+    actual: iterator.next().value,
+    expected: put(Actions.NotificationBar.onShowNotificationBar({
+      type: 'success',
+      message: `Success! Your Work ID is ${workId}. Redirecting to Explorer-web...`,
```

Comment:
Not sure what the intention is here in the UI, but we don't really want to redirect them. There should be a link they can click instead, which opens the Works Explorer in a new tab. Let's discuss before you implement it.